### PR TITLE
[enh] oscar: image thumbnail layout

### PR DESCRIPTION
**Pull request by dalf**
## What does this PR do?

* Adjust thumbnail sizes to fill the container width (code copied and adjusted from the simple theme).
* Remove border on thumbnails

#### With javascript
image

image

#### Without javascript
image

## Why is this change important?

Improve UX.

## How to test this PR locally?

use the different oscar styles, search for
* `!images time`
* `!general !images time`

Try javascript enable and disabled.

### CODE REVIEW

File: searx/static/themes/oscar/js/searx.js [hunk 1]
```diff
@@ -199,7 +202,175 @@ $(document).ready(function(){
...
         $(a.target).parent().attr("aria-selected", "true");
     });
+
+    /**
+     * Layout images according to their sizes
+     */
+    searx.image_thumbnail_layout = new searx.ImageLayout('#main_results', '#main_results .result-images', 'img.img-thumbnail', 200);
+    searx.image_thumbnail_layout.watch();
 });
+;/**
```
Comment (return42) [inline searx/static/themes/oscar/js/searx.js]: I guess this is a code duplication of  [`searx/static/themes/oscar/src/js/imageresult.js`](https://github.com/searx/searx/pull/2675/files#diff-3051b1de9173fce6d86a7260a2462e9ab106608ae39f9148ca0bb9b043f037a3)?

**Review by return42 — DISMISSED** [review]
Looks much better / Thank you!

What do you think about the code duplication?

File: searx/static/themes/oscar/js/searx.js (hunk 1 above)
Comment (dalf) [reply searx/static/themes/oscar/js/searx.js]: Yes 
https://github.com/searx/searx/blob/46db9a8600f1aec5c0780b0248fb6476c1ad5078/searx/static/themes/oscar/src/js/imageresult.js
is nearly duplicated of
https://github.com/searx/searx/blob/46db9a8600f1aec5c0780b0248fb6476c1ad5078/searx/static/themes/simple/js/searx_src/searx_imageresult.js

I'm going to keep only one file.

Comment (dalf) [reply searx/static/themes/oscar/js/searx.js]: Updated.

Now, using grunt, [`searx/static/themes/__common__/js/image_layout.js`](https://github.com/searx/searx/blob/b9e8ed063641d168e1a8318a5a07457f8e05e957/searx/static/themes/__common__/js/image_layout.js) is embedded in:
* `searx/static/themes/oscar/js/searx.min.js`
* `searx/static/themes/simple/js/searx.min.js`

File: searx/static/themes/__common__/js/image_layout.js [hunk 2]
```diff
@@ -65,38 +66,52 @@
...
-          continue w;
+          isSearching = false;
         }
       }
-      this._setSize(slice, Math.min(this.maxHeight, h));
-      break;
+      if (isSearching) {
+        this._setSize(slice, Math.min(this.maxHeight, h));
+        break;
+      }
```
Comment (dalf) [inline searx/static/themes/__common__/js/image_layout.js:99]: `continue w` is replaced by `isSearching` : try to make the code more readable.

File: searx/static/themes/oscar/css/logicodev-dark.css [hunk 3]
```diff
@@ -202,14 +202,39 @@ input[type=checkbox]:not(:checked) + .label_hide_if_checked + .label_hide_if_not
...
+}
+.result-images a:hover,
+.result-images a:focus {
+  outline: 0;
+}
+.result-images a:hover .img-thumbnail,
+.result-images a:focus .img-thumbnail {
+  box-shadow: 5px 5px 15px 0px black;
+}
+.result-images.js a .img-thumbnail {
```
Comment (dalf) [inline searx/static/themes/oscar/css/logicodev-dark.css:225]: `ImageLayout` adds the `js` class for each processed image.

**Review by kvch — DISMISSED** [review]
After a rebase it is good to go.

**Review by BBaoVanC — APPROVED** [review]
Can this be merged soon? I've had this PR merged on my fork for a little over a week now and it works great!

**Review by asciimoo — APPROVED** [review]
Great, thanks!

**Pull request merged**